src/codec/decoders.rs

```diff
@@ -0,0 +1,23 @@
+use vector_core::event::Event;
+
+pub use crate::config::decoding::{Decoding, DecodingConfig};
+
+pub enum Decoder {
+    Utf8,
+    Json,
+}
+
+impl Decoder {
+    pub fn decode(self, event: Event) -> Event {
+        event
```

Comment:
If we merge this in before we implement this bit, could we add a comment here making it clear that this is not implemented yet?

Reply:
Yep!